trackma/hooks.py

```diff
@@ -0,0 +1,33 @@
+__path__ = []
+
+
+def iter():
+    """Return an iterator over the modules in __path__."""
+    if not __path__:
+        return ()
+
+    from pkgutil import iter_modules
+    return (name for _, name, _ in iter_modules(__path__))
+
+
+def load(name, attr=None):
+    if not __path__:
+        raise ImportError(f"Cannot load hook '{name}'")
+
+    from importlib import import_module
+    return import_module(f'trackma.hooks.{name}')
```

Comment:
Using `iter_modules` to iterate modules in a folder, only to throw away all but one part of the return values, still seems like a waste to me, but I also realize that using `import_module` and the previous way of managing hooks has different (side) effects. Notably I assume that the modules were not put into `sys.modules` before, which may or may not be desirable. Building a module path string can also be fragile in the sense that `iter_modules` might return module names that cannot be imported like this (e.g. when they contain hyphens or spaces), but I don't actually know about this. Users should not use such characters in their hook file names either way.

Reply:
We can urlencode the name and replace `%` with `_`, or throw an error if hook name contains invalid characters?

I'd prefer this approach (turning `trackma.hooks` into a namespace package), cause now I can share code between hooks (I have a `auth` hook, that only provides `init/destroy/get_creds`, and keeps credentials in libsecret, that is used by other hooks)

Reply:
nvm, python doesn't enforce valid module name if using `import_module`, only disallowed character seems to be `.`

```
[D] Engine: Importing hook local test...
<module 'trackma.hooks.local test' from '/home/ahm/.config/trackma/hooks/local test.py'>
[D] Engine: Importing hook local-test...
<module 'trackma.hooks.local-test' from '/home/ahm/.config/trackma/hooks/local-test.py'>
```